torch/csrc/jit/codegen/cuda/test/test_gpu_outer_reduction.cpp

```diff
@@ -221,6 +219,1125 @@ TEST_F(NVFuserTest, FusionGroupedGridWelfordOuterOpt_CUDA) {
+  } else if (params.unswitch == 1) {
+    reduction_tv->split(unswitch_pos, 1, false);
+    reduction_tv->axis(unswitch_pos)->parallelize(ParallelType::Unswitch);
+  }
+
+  // Parallelize the iteration domain with vectorization, TIDx, and BIDx
+  reduction_tv->split(-1, params.vec);
+  reduction_tv->axis(-1)->parallelize(ParallelType::Vectorize);
+  reduction_tv->split(-2, params.tidx);
+  reduction_tv->axis(-2)->parallelize(ParallelType::TIDx);
+  reduction_tv->split(-3, params.bidx);
+  reduction_tv->axis(-3)->parallelize(ParallelType::BIDx);
+
+  auto reduction_tv_rf =
+      reduction_scheduler_utils::sortAndRFactor(reduction_tv);
+
+  // Make sure the vectorized domain placed at the innermost position
+  int vec_id_cur_pos = -1;
+  std::unordered_map<int, int> vec_reorder_map;
+  for (const auto i : c10::irange(reduction_tv_rf->nDims())) {
+    auto id = reduction_tv_rf->axis(i);
+    if (id->getParallelType() == ParallelType::Vectorize) {
+      vec_id_cur_pos = i;
+      vec_reorder_map[i] = -1;
+    } else if (vec_id_cur_pos >= 0) {
+      vec_reorder_map[i] = i - 1;
+    }
+  }
+  TORCH_CHECK(vec_id_cur_pos != -1, "Vectorized ID not found");
+  reduction_tv_rf->reorder(vec_reorder_map);
+
+  TransformPropagator propagator(reduction_tv_rf);
+  MaxRootDomainInfoSpanningTree(reduction_tv_rf).traverse(&propagator);
+
+  // Clear vectorization and unswitch as we want to selectively use
+  // them
+
+  // Clear the vectorization
+  reduction_tv->axis(-1)->parallelize(ParallelType::Serial);
+  reduction_tv_rf->axis(-1)->parallelize(ParallelType::Serial);
+
+  // Clear unswitch
+  IterDomain* unswitch_id = nullptr;
+  auto unswitch_id_it = std::find_if(
+      reduction_tv_rf->domain()->domain().begin(),
+      reduction_tv_rf->domain()->domain().end(),
+      [](auto id) { return id->getParallelType() == ParallelType::Unswitch; });
+  if (unswitch_id_it != reduction_tv_rf->domain()->domain().end()) {
+    unswitch_id = *unswitch_id_it;
+  }
+
+  if (unswitch_id != nullptr) {
+    unswitch_id->parallelize(ParallelType::Serial);
+  }
+
+  scheduler_utils::parallelizeAllLike(
+      reduction_tv_rf, ir_utils::allTvs(&fusion));
+
+  // Vectorize inputs
+  for (auto input_cache : input_caches) {
+    // Skip if the maximum supported size is exceeded
+    if (input_cache->getDataType() == DataType::Float && params.vec == 8) {
+      continue;
+    }
+    input_cache->axis(-1)->parallelize(ParallelType::Vectorize);
+  }
+
+  // Vectorize outputs
+  for (auto output : ir_utils::filterByType<TensorView>(fusion.outputs())) {
+    // Skip if the maximum supported size is exceeded
+    if (output->getDataType() == DataType::Float && params.vec == 8) {
+      continue;
+    }
+    output->axis(-1)->parallelize(ParallelType::Vectorize);
+  }
+
+  // Cross-iteration grouping of grid reductions
+  for (auto reduction_tv : reduction_tvs) {
+    reduction_tv->axis(-1)->parallelize(ParallelType::Group);
+  }
```

Comment:
For the parallelization above should you be using `scheduler_utils::parallelizeAllLike` with the `selected_tvs` option?